finagle-http/src/main/scala/com/twitter/finagle/http/filter/StatsFilter.scala

```diff
@@ -23,6 +23,12 @@ class StatsFilter[REQUEST <: Request](stats: StatsReceiver)
   private[this] val timeReceiver = stats.scope("time")
   private[this] val responseSizeStat = stats.stat("response_size")
 
+  private[this] val counterCache: Map[String, Counter] =
+    Map.empty.withDefault(statusReceiver.counter(_))
```

Comment:
lol double whammy.  @spockz let's go ahead with your original plan and just use a map.